test/e2e/performanceprofile/functests/2_performance_update/tuned_deferred.go

```diff
@@ -0,0 +1,320 @@
+
+					cfg, err := ini.Load(tunedData)
+					Expect(err).ToNot(HaveOccurred())
+
+					sysctlSection, err := cfg.GetSection("sysctl")
+					Expect(err).ToNot(HaveOccurred())
+
+					sysctlSection.Key("kernel.shmmni").SetValue(tc.Kernel_shmmni)
+					Expect(sysctlSection.Key("kernel.shmmni").String()).To(Equal(tc.Kernel_shmmni))
+
+					var updatedTunedData bytes.Buffer
+					_, err = cfg.WriteTo(&updatedTunedData)
+					Expect(err).ToNot(HaveOccurred())
+
+					updatedTunedDataBytes := updatedTunedData.Bytes()
+					tuned.Spec.Profile[0].Data = stringPtr(string(updatedTunedDataBytes))
+
+					testlog.Infof("Updating Tuned Profile Patch: %s", name)
+					Expect(testclient.ControlPlaneClient.Update(context.TODO(), tuned)).To(Succeed())
+				}
+
+				switch tc.ExpectedBehavior {
+				case "immediate":
+					verifyImmediateApplication(tuned, workerCNFNodes, tc.Kernel_shmmni)
+				case "deferred":
+					out, err := verifyDeferredApplication(tuned, workerCNFNodes, tc.Kernel_shmmni, name)
+					Expect(err).To(BeNil())
+					Expect(out).To(BeTrue())
+					rebootNode(poolName)
+					verifyImmediateApplication(tuned, workerCNFNodes, tc.Kernel_shmmni)
+				}
+			},
+			Entry("[test_id:78115] Verify deferred Always with first-time profile change", TunedProfileConfig{
+				DeferMode:         "always",
+				ProfileChangeType: "first-time",
+				ExpectedBehavior:  "deferred",
+				Kernel_shmmni:     "8192",
+			}),
+			Entry("[test_id:78116] Verify deferred Always with in-place profile update", TunedProfileConfig{
+				DeferMode:         "always",
+				ProfileChangeType: "in-place",
+				ExpectedBehavior:  "deferred",
+				Kernel_shmmni:     "4096",
+			}),
+			Entry("[test_id:78117] Verify deferred Update with first-time profile change", TunedProfileConfig{
+				DeferMode:         "update",
+				ProfileChangeType: "first-time",
+				ExpectedBehavior:  "immediate",
+				Kernel_shmmni:     "8192",
+			}),
+			Entry("[test_id:78118] Verify deferred Update with in-place profile update", TunedProfileConfig{
+				DeferMode:         "update",
+				ProfileChangeType: "in-place",
+				ExpectedBehavior:  "deferred",
+				Kernel_shmmni:     "4096",
+			}),
+			Entry("[test_id:78119] Verify deferred with No annotation with first-time profile change", TunedProfileConfig{
+				DeferMode:         "",
+				ProfileChangeType: "first-time",
+				ExpectedBehavior:  "immediate",
+				Kernel_shmmni:     "8192",
+			}),
+			Entry("[test_id:78120] Verify deferred Never mode with in-place profile update", TunedProfileConfig{
+				DeferMode:         "never",
+				ProfileChangeType: "in-place",
+				ExpectedBehavior:  "immediate",
+				Kernel_shmmni:     "4096",
+			}),
+		)
+	})
+
+	AfterAll(func() {
+		tuned = getTunedProfile(name)
+		if tuned != nil {
+			Expect(testclient.ControlPlaneClient.Delete(context.TODO(), tuned)).To(Succeed())
+		}
+	})
+
+})
+
+func stringPtr(s string) *string {
+	return &s
+}
+
+func getTunedProfile(tunedName string) *tunedv1.Tuned {
```

Comment:
do we need `ginkgo.GinkgoHelper()` here?

Reply:
Added, Thanks